pkgs/test_api/lib/src/frontend/expect_async.dart

```diff
@@ -176,7 +176,7 @@ class _ExpectedFunction<T> {
       _run([a0, a1, a2, a3, a4, a5].where((a) => a != placeholder));
 
   /// Runs the wrapped function with [args] and returns its return value.
-  T _run(Iterable args) {
+  T _run(Iterable<Object?> args) {
```

Comment:
Why is this not dynamic here?

Reply:
I think that we do not need to use dynamic dispatch, so `Object?` is a safer type to use. But I can change to `dynamic` if that is preferred for this repository.

Reply:
We should go with `Object?` unless we really want dynamic in a specific place for some reason